receiver/awscontainerinsightreceiver/internal/gpu/decorator.go

```diff
@@ -0,0 +1,204 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gpu
+
+import (
+	"context"
+	"errors"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+)
+
+const (
+	gpuUtil        = "DCGM_FI_DEV_GPU_UTIL"
+	gpuMemUtil     = "DCGM_FI_DEV_FB_USED_PERCENT"
+	gpuMemUsed     = "DCGM_FI_DEV_FB_USED"
+	gpuMemTotal    = "DCGM_FI_DEV_FB_TOTAL"
+	gpuTemperature = "DCGM_FI_DEV_GPU_TEMP"
+	gpuPowerDraw   = "DCGM_FI_DEV_POWER_USAGE"
+)
+
+var _ stores.CIMetric = (*gpuMetric)(nil)
+
+var metricToUnit = map[string]string{
+	gpuUtil:        "Percent",
+	gpuMemUtil:     "Percent",
+	gpuMemUsed:     "Bytes",
+	gpuMemTotal:    "Bytes",
+	gpuTemperature: "None",
+	gpuPowerDraw:   "None",
+}
+
+type gpuMetric struct {
```

Comment:
Yeah, great idea. Renamed `CAdvisorMetric` to `RawContainerInsightsMetric` which doesn't seem to be the best but it is what really that is. Scanning thru the code, `cgroupPath` is only being set with container name, and it now updated to `ContainerName`. I'm thinking to just leave it as blank for GPU metric processing since there will be no merge of 2 metrics (at least for now).